src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php

```diff
@@ -435,7 +435,14 @@ public function setRequiredFieldsAction(Request $request)
     public function searchAction(Request $request)
     {
         $query = $request->query->get('customer_search');
-        $phrases = explode(' ', $query);
+        $configuration = $this->get('prestashop.adapter.legacy.configuration');
+
+        if (!$configuration->get('PS_CUSTOMER_SEARCH_FULL')) {
+            $phrases = explode(' ', $query);
```

Comment:
Maybe we should add a placeholder on the page and use it as a separator. Here we are choosing the white space as the one and only 'separator' character that can be used.

Maybe some people would expect the separator character to be ; or , or / . We just need to make sure there is no confusion, it's not ambiguous.

Reply:
@matks The "space" separator is just the original behavior that just sucks... 😄😄 I would just change it to searching the whole word, but @okom3pom tried to be nice and keep the option of original behavior, if anyone wanted to do it. I would not invest any time developing this feature.

I think all people will change to our new behavior right away, I search for Mathieu Ferment and it will show me all Mathieus and all Ferments, completely unusable.

Reply:
My proposal - original behavior for people who upgrade, new behavior for new users.